src/extendedstorage.cpp

```diff
@@ -738,11 +735,28 @@ void ExtendedStorage::Private::clearAlarms(const Incidence::Ptr &incidence)
     }
 }
 
-void ExtendedStorage::Private::clearAlarms(const KCalendarCore::Incidence::List &incidences)
+void ExtendedStorage::Private::clearAlarms(const Incidence::List &incidences,
+                                           const Calendar::Ptr &calendar)
 {
-    foreach (const Incidence::Ptr incidence, incidences) {
+    QSet<QString> recurringUids;
+    for (const Incidence::Ptr &incidence : incidences) {
+        if (incidence->recurs()) {
+            recurringUids.insert(incidence->uid());
+        }
+    }
+    Incidence::List recurring;
+    for (const Incidence::Ptr incidence : incidences) {
         clearAlarms(incidence);
+        if (incidence->hasRecurrenceId() && !recurringUids.contains(incidence->uid())) {
+            Incidence::Ptr parent = calendar->incidence(incidence->uid());
+            if (parent) {
+                clearAlarms(parent);
+                recurring << parent;
+                recurringUids.insert(parent->uid());
+            }
+        }
     }
+    setAlarms(recurring, calendar);
```

Comment:
This somehow bothers me a bit when clearAlarms() sets alarms. Not that I have alternatives to suggest.

Reply:
I'm pushing an additionnal commit with a proposition to remove the clearAlarms(list, calendar) function and put it where it is used since it was used only in one place.

Reply:
Hey that's very nice. The code suddenly became a lot easier to follow. Could you still squash that commit to the first one.

One thing that was also confusing me was the overloaded methods that modified either a single instance, a list or a notebook. That's now almost all gone. For the remaining clearAlarms(notebook_uid) could consider would it be more readable as clearNotebookAlarms() or something, but up to you, just a related suggestion.

Reply:
I've squashed the commits and renamed the `clearAlarms(notebookUid)` into `clearAlarmsByNotebook(notebookUid)`.